lib/_tls_wrap.js

```diff
@@ -672,51 +668,25 @@ TLSSocket.prototype.getPeerCertificate = function(detailed) {
   return null;
 };
 
-TLSSocket.prototype.getFinished = function() {
-  if (this._handle)
-    return this._handle.getFinished();
-};
-
-TLSSocket.prototype.getPeerFinished = function() {
-  if (this._handle)
-    return this._handle.getPeerFinished();
-};
-
-TLSSocket.prototype.getSession = function() {
-  if (this._handle) {
-    return this._handle.getSession();
-  }
-
-  return null;
-};
-
-TLSSocket.prototype.isSessionReused = function() {
-  if (this._handle) {
-    return this._handle.isSessionReused();
-  }
-
-  return null;
-};
-
-TLSSocket.prototype.getCipher = function(err) {
-  if (this._handle) {
-    return this._handle.getCurrentCipher();
-  } else {
+// Proxy TLSSocket handle methods
+function makeSocketMethodProxy(name) {
+  return function socketMethodProxy(...args) {
```

Comment:
Won't this show up in stack traces now instead of the actual proxied function name? I think having the actual function name in the stack trace would be more helpful.

Reply:
V8 is able to infer the method names. This should show up as `TLSSocket. socketMethodProxy <as getFinished>` (previously `TLSSocket.getFinished`)

Reply:
Yep, like @joyeecheung is saying, if an error was thrown in `getFinished` it would now appear as `at TLSSocket.socketMethodProxy [as getFinished]`